### ISSUE TITLE
Exception during table parsing when borders are not aligned

### ISSUE THREAD

**Issue by radoslav006** [python-openxml/python-docx#1146]
The exception is raised when parsing table with not aligned borders.

Library version: 0.8.11

Table picture:
![obraz](https://user-images.githubusercontent.com/1621331/197142892-225b6f8b-fa52-4c75-b8c4-66cf428220e8.png)

Xml:
[err.xml.txt](https://github.com/python-openxml/python-docx/files/9836882/err.xml.txt)

Code:
```python
from docx import Document


if __name__ == '__main__':
    doc = Document('doc.docx')
    for table in doc.tables:
        for row in table.rows:
            for cell in row.cells:
                print(cell.text)
```

Exception:
```
  File "C:\Python39\lib\site-packages\docx\table.py", line 173, in _cells 
    cells.append(cells[-col_count])
IndexError: list index out of range
```

**Comment by djplaner** [comment]
FWIW, I've stumbled across a similar issue due to the same bad practice in the HTML (rows where the number of cells in the row exceeded the initial default)

The problem appears to be that the ```cell``` function in the [tables.py](https://github.com/python-openxml/python-docx/blob/master/docx/table.py) file doesn't check if the cell exists before trying to access it.  I've modified my local version to the following.

Check the index doesn't exceed the array, if it does return None.  Depending on how you're using this, you may need to make further changes upstream.

```
    def cell(self, row_idx, col_idx):
        """
        Return |_Cell| instance correponding to table cell at *row_idx*,
        *col_idx* intersection, where (0, 0) is the top, left-most cell.
        """
        cell_idx = col_idx + (row_idx * self._column_count)
        if cell_idx >= len(self._cells):
            return None
        return self._cells[cell_idx]
```

**Comment by tonal** [comment]
try https://github.com/python-openxml/python-docx/issues/881#issuecomment-1437979336

**Comment by radoslav006** [comment]
> try [#881 (comment)](https://github.com/python-openxml/python-docx/issues/881#issuecomment-1437979336)

Thanks for a workaround, maybe someone find it helpful, but in the end this need to be fixed within the library.

**Comment by scanny** [comment]
Fixed in v1.1.1 circa Apr 30, 2024.
